source/templates/Pict-Template-DataWithTemplateFallback.js

```diff
@@ -0,0 +1,151 @@
+const libPictTemplate = require('pict-template');
+
+class PictTemplateProviderData extends libPictTemplate
+{
+	/**
+	 * @param {Object} pFable - The Fable Framework instance
+	 * @param {Object} pOptions - The options for the service
+	 * @param {String} pServiceHash - The hash of the service
+	 */
+	constructor(pFable, pOptions, pServiceHash)
+	{
+		super(pFable, pOptions, pServiceHash);
+
+		/** @type {any} */
+		this.log;
+
+		this.addPattern('{~DataWithTemplateFallback:', '~}');
+		this.addPattern('{~DWTF:', '~}');
+	}
+
+	/**
+	 * Render a template expression, returning a string with the resulting content or an optional default template string from an address.
+	 *
+	 * @param {string} pTemplateHash - The hash contents of the template (what's between the template start and stop tags)
+	 * @param {any} pRecord - The json object to be used as the Record for the template render
+	 * @param {Array<any>} pContextArray - An array of context objects accessible from the template; safe to leave empty
+	 *
+	 * @return {string} The rendered template
+	 */
+	render(pTemplateHash, pRecord, pContextArray)
+	{
+		let tmpHash = pTemplateHash.trim();
+		let tmpRecord = (typeof(pRecord) === 'object') ? pRecord : {};
+
+		if (this.pict.LogNoisiness > 4)
+		{
+			this.log.trace(`PICT DataWithTemplateFallback [fDataRender]::[${tmpHash}] with tmpData:`, tmpRecord);
+		}
+		else if (this.pict.LogNoisiness > 3)
+		{
+			this.log.trace(`PICT DataWithTemplateFallback [fDataRender]::[${tmpHash}]`);
+		}
+
+		let tmpTemplateFallbackAddress = '';
+		if (tmpHash.indexOf(':') > -1)
+		{
+			tmpTemplateFallbackAddress = tmpHash.split(':')[1];
+			tmpHash = tmpHash.split(':')[0];
+		}
+
+		let tmpValue = '';
+		if (tmpHash != null)
+		{
+			tmpValue = this.resolveStateFromAddress(tmpHash, tmpRecord, pContextArray);
+
+			if (tmpValue && tmpValue !== 'undefined')
+			{
+				if (this.pict.LogNoisiness > 3)
+				{
+					this.log.trace(`PICT DataWithTemplateFallback [fDataRender]::[${tmpHash}] - Found value: ${tmpValue}`);
+					// If a value is found, return it
+					return tmpValue;
+				}
+				return tmpValue;
```

Comment:
This return from inside the if, while not strictly "wrong" is weird and confusing. Should remove it.